src/lib.rs

```diff
@@ -28,9 +28,51 @@
 //! Generally usage of this api is error-prone and discouraged if not needed.
 //!
 //! ## Getting started
-//!
 //! To get started check out the [`Libinput` struct](./struct.Libinput.html).
 //!
+//! Here's a small example that prints all events:
+//! ```
+//! extern crate input;
+//! use input::{Libinput, LibinputInterface};
+//! use std::fs::{File, OpenOptions};
+//! use std::os::unix::fs::OpenOptionsExt;
+//! use std::os::unix::io::{RawFd, FromRawFd, IntoRawFd};
+//! use std::path::Path;
+//! use udev::Context;
+//! 
+//! extern crate libc;
+//! use libc::{O_RDONLY, O_RDWR, O_WRONLY};
+//! 
+//! struct Interface;
+//! 
+//! impl LibinputInterface for Interface {
+//! 	fn open_restricted(&mut self, path: &Path, flags: i32) -> Result<RawFd, i32> {
+//! 		OpenOptions::new()
+//! 		    .custom_flags(flags)
+//! 		    .read((flags & O_RDONLY != 0) | (flags & O_RDWR != 0))
+//! 		    .write((flags & O_WRONLY != 0) | (flags & O_RDWR != 0))
+//! 			.open(path)
+//! 			.map(|file| file.into_raw_fd())
+//! 			.map_err(|err| err.raw_os_error().unwrap())
+//! 	}
+//! 	fn close_restricted(&mut self, fd: RawFd) {
+//! 		unsafe {
+//! 			File::from_raw_fd(fd);
+//! 		}
+//! 	}
+//! }
+//! 
+//! fn main() {
+//! 	let mut input = Libinput::new_from_udev(Interface {}, &Context::new().unwrap());
+//! 	input.udev_assign_seat("seat0").unwrap();
+//! 	loop {
+//! 		input.dispatch().unwrap();
+//! 		while let Some(event) = input.next() {
```

Comment:
I personally like using a for loop better:
```suggestion
//! 		for even in &mut input {
```

Reply:
Thanks, I had a lot of trouble iterating over the events without owning `input`, this was the first solution that worked for me.